use-cases/using-Intel-QAT-and-kata.md

```diff
@@ -0,0 +1,484 @@
+[https://github.com/kata-containers/packaging/tree/master/kernel](https://
+github.com/kata-containers/packaging/tree/master/kernel). The config for the 
+kata-kernel is at 
+[https://github.com/kata-containers/packaging/tree/master/kernel/configs](
+https://github.com/kata-containers/packaging/tree/master/kernel/configs) In 
+this example, the latest kernel supported by Kata is automatically used.
+There are some patches that must be installed as well which the 
+build-kernel.sh script should automatically apply, but if you are using a 
+different kernel version then they may need to be manually applied. Since the 
+Kata Containers kernel has a minimal set of kernel flags set, the config must 
+be updated to set some of the `CRYPTO` and `STACKPROTECTOR` flags to enabled. 
+This may change with different kernel versions. The following was tested with 
+kernel `v4.19.28-41`.
+
+```sh
+$ mkdir -p $GOPATH
+$ cd $GOPATH
+$ go get -v github.com/kata-containers/packaging
+$ $GOPATH/src/github.com/kata-containers/packaging/kernel/build-kernel.sh setup
+```
+
+### Build Kata Kernel
+
+```sh
+$ export LINUX_VER=$(ls -d kata*)
+$ sed -i 's/EXTRAVERSION =/EXTRAVERSION = .qat.container/' $LINUX_VER/Makefile
+
+# Change kernel config options to support QAT
+$ sed -i 's/# CONFIG_PCI_IOV is not set/CONFIG_PCI_IOV=Y/' $LINUX_VER/.config
+$ sed -i 's/# CONFIG_CRYPTO_SHA1 is not set/CONFIG_CRYPTO_SHA1=Y/' $LINUX_VER/.config
+$ sed -i 's/# CONFIG_CRYPTO_DH is not set/CONFIG_CRYPTO_DH=Y/' $LINUX_VER/.config
+$ sed -i 's/# CONFIG_STACKPROTECTOR is not set/CONFIG_STACKPROTECTOR=Y/' $LINUX_VER/.config
+$ $GOPATH/src/github.com/kata-containers/packaging/kernel/build-kernel.sh build
+```
+
+> **Note: There may be additional kernel parameters that need to be set to yes 
+> when compiling.**
+
+### Copy Kata Kernel
+
+```sh
+$ mkdir -p $KATA_KERNEL_LOCATION
+$ cp $LINUX_VER/arch/x86/boot/bzImage $KATA_KERNEL_LOCATION/vmlinuz-${LINUX_VER}_qat
+```
+
+### Prepare Kata root filesystem
+
+These instructions build upon the OS builder instructions located in the  
+[Developer Guide](../Developer-Guide.md). This example will use Clear Linux 
+(Kata Containers default) as the root filesystem with systemd as the init and 
+will add in the `kmod` binary which is not a standard binary in a Kata image.  
+Docker should already be installed. If a custom kata-agent needs to be used 
+please refer to the link above on how to add in.
+
+```sh
+$ mkdir -p $OSBUILDER
+$ cd $OSBUILDER
+$ git clone https://github.com/kata-containers/osbuilder.git
+$ export ROOTFS_DIR=${OSBUILDER}/osbuilder/rootfs-builder/rootfs
+$ export EXTRA_PKGS='kmod'
+
+# Make Sure that the kata agent version matches the installed kata-runtime version
+$ export AGENT_VERSION=$(kata-runtime version | head -n 1 | grep -o "[0-9.]\+")
+$ cd ${OSBUILDER}/osbuilder/rootfs-builder
+$ sudo rm -rf ${ROOTFS_DIR}
+$ script -fec 'sudo -E GOPATH=$GOPATH USE_DOCKER=true SECCOMP=no ./rootfs.sh clearlinux'
```

Comment:
You can add `EXTRA_PKGS='kmod'` here, rather than exporting it earlier.

Reply:
I know. I purposefully put it above with the other export commands for readability.